src/test/java/org/jenkinsci/plugins/credentialsbinding/masking/BashPatternMaskerProviderTest.java

```diff
@@ -0,0 +1,121 @@
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.experimental.theories.DataPoint;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.startsWith;
+import static org.jenkinsci.plugins.credentialsbinding.test.Executables.executable;
+import static org.junit.Assume.assumeThat;
+
+@RunWith(Theories.class)
+public class BashPatternMaskerProviderTest {
+
+    public static final @DataPoint String SAMPLE_PASSWORD = "}#T14'GAz&H!{$U_";
+    public static final @DataPoint String ANOTHER_SAMPLE_PASSWORD = "a'b\"c\\d(e)#";
+    public static final @DataPoint String ONE_MORE = "'\"'(foo)'\"'";
+
+    @DataPoints
+    public static List<String> generatePasswords() {
+        Random random = new Random(100);
+        List<String> passwords = new ArrayList<>(10);
+        for (int i = 0; i < 10; i++) {
+            int length = random.nextInt(24) + 8;
+            StringBuilder sb = new StringBuilder(length);
+            for (int j = 0; j < length; j++) {
+                char next = (char) (' ' + random.nextInt('\u007f' - ' ')); // 0x7f is DEL, 0x7e is ~, and space is the first printable ASCII character
+                sb.append(next);
+            }
+            passwords.add(sb.toString());
+        }
+        return passwords;
+    }
+
+    @Rule public JenkinsRule j = new JenkinsRule();
+
+    private WorkflowJob project;
+    private String credentialsId;
+
+    @BeforeClass
+    public static void assumeBash() {
+        assumeThat("bash", is(executable()));
+        // due to https://github.com/jenkinsci/durable-task-plugin/blob/e75123eda986f20a390d92cc892c3d206e60aefb/src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java#L149
+        // on Windows
+        assumeThat("nohup", is(executable()));
+    }
+
+    @Before
+    public void setUp() throws IOException {
+        j.jenkins.getDescriptorByType(Shell.DescriptorImpl.class).setShell(Executables.getPathToExecutable("bash"));
+        project = j.createProject(WorkflowJob.class);
+        credentialsId = UUID.randomUUID().toString();
+        project.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "  withCredentials([string(credentialsId: '" + credentialsId + "', variable: 'CREDENTIALS')]) {\n" +
+                        "    sh ': \"$CREDENTIALS\"'\n" + // : will expand its parameters and do nothing with them
+                        "    sh ': \"< $CREDENTIALS >\"'\n" +
```

Comment:
What is the significance of the `<`…`>` here? (a comment would be in order)

Reply:
Oh, nothing more significant than using "before1" and "after1", just less typing.

Reply:
Adding docs. Other resolved things are likewise being addressed.